pkg/controller/volume/attachdetach/cache/actual_state_of_world.go

```diff
@@ -234,6 +236,20 @@ func (asw *actualStateOfWorld) MarkVolumeAsDetached(
 	asw.DeleteVolumeNode(volumeName, nodeName)
 }
 
+func (asw *actualStateOfWorld) MarkDesireToDetachVolume(
+	volumeName api.UniqueVolumeName, nodeName string) error {
+	asw.Lock()
+	defer asw.Unlock()
+	return asw.removeVolumeFromReportAsAttached(volumeName, nodeName)
+}
+
+func (asw *actualStateOfWorld) UnmarkDesireToDetachVolume(
```

Comment:
These names are now misleading. Let's change `UnmarkDesireToDetachVolume` to `AddVolumeToReportAsAttached`.